frontend/src/views/Lobby.tsx

```diff
@@ -337,19 +343,64 @@ function LobbyPage() {
       conditionallyBootKickedUser(room, currentUser);
     };
 
+    setLoading(true);
     connect(roomId, userId).then(() => {
       // Body encrypt through JSON.
       subscribe(routes(roomId).subscribe_lobby, subscribeCallback).then((subscriptionParam) => {
         setSubscription(subscriptionParam);
         setSocketConnected(true);
+        setError('');
+        setLoading(false);
       }).catch((err) => {
         setError(err.message);
+        setLoading(false);
       });
     }).catch((err) => {
       setError(err.message);
+      setLoading(false);
     });
   }, [currentUser, conditionallyBootKickedUser]);
 
+  const refreshRoomDetails = () => {
+    // Call GET endpoint to get latest room info
+    if (!loading) {
+      setLoading(true);
+      getRoom(location.state.roomId)
+        .then((res) => {
+          setStateFromRoom(res);
+
+          // Boot the user from the room, if they are not present.
+          updateCurrentUserDetails(res.users);
+
+          // Attempt to connect the user to the socket.
+          if (currentUser && currentUser.userId) {
+            connectUserToRoom(res.roomId, currentUser.userId);
+          }
+        })
+        .catch((err) => setError(err));
+    }
+  };
+
+  const leaveRoom = () => {
+    // eslint-disable-next-line no-alert
+    if (window.confirm('Are you sure you want to leave the room?')) {
+      if (currentUser && currentUser.userId) {
+        setLoading(true);
+        setError('');
+        removeUser(currentRoomId, {
+          initiator: currentUser,
+          userToDelete: currentUser,
+        });
+        disconnect();
```

Comment:
I might prefer to have the redirect and disconnect moved into the then block of removeUser just for safety's sake (with async stuff if it redirects you from the page halfway through the removeUser axios call, idk if that would be prone to occasional misses) - but if you've tested this a lot and it works every time, I think that should be okay too

Reply:
I do think that we can leave this in - this doesn't have them leave the whole website, so this code still runs, and I haven't encountered any issues thus far. I'm happy to change this back if we do encounter that, but thus far even if it redirects and disconnects the user before completing the removeUser call, that call still comes back. Since I think it should redirect and disconnect the user either way, I think it makes sense to include it regardless of removeUser's success as well.